ABDefinitionCore.js

```diff
@@ -42,6 +42,17 @@ module.exports = class ABDefinitionCore {
       }
       this.type = attributes.type || attributes?.json?.type || "";
       this.json = attributes.json || null;
+
+      this.json.versionData = attributes.json?.versionData || {
```

Comment:
This is what you are already doing in `ABApplicationCore`.  We don't need to make this assignment here as well.